src/test/java/org/jabref/logic/xmp/XMPUtilWriterTest.java

```diff
@@ -0,0 +1,162 @@
+import org.jabref.logic.importer.ParseException;
+import org.jabref.logic.importer.fileformat.BibtexParser;
+import org.jabref.model.entry.BibEntry;
+import org.jabref.model.util.DummyFileUpdateMonitor;
+import org.jabref.model.util.FileUpdateMonitor;
+
+import org.apache.pdfbox.pdmodel.PDDocument;
+import org.apache.pdfbox.pdmodel.PDPage;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.TemporaryFolder;
+import org.mockito.Answers;
+
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+public class XMPUtilWriterTest {
+
+    @Rule public TemporaryFolder tempFolder = new TemporaryFolder();
+
+    private static final FileUpdateMonitor fileMonitor = new DummyFileUpdateMonitor();
+
+    private XMPPreferences xmpPreferences;
+
+    private ImportFormatPreferences importFormatPreferences;
+
+    private BibtexParser parser;
+
+    private static final String olly2018 = "@article{Olly2018,\r\n" +
+            "  author       = {Olly and Johannes},\r\n" +
+            "  title        = {Stefan's palace},\r\n" +
+            "  journal      = {Test Journal},\r\n" +
+            "  volume       = {1},\r\n" +
+            "  number       = {1},\r\n" +
+            "  pages        = {1-2},\r\n" +
+            "  month        = mar,\r\n" +
+            "  issn         = {978-123-123},\r\n" +
+            "  note         = {That's a note},\r\n" +
+            "  abstract     = {That's an abstract},\r\n" +
+            "  comment      = {That's a comment},\r\n" +
+            "  doi          = {10/3212.3123},\r\n" +
+            "  file         = {:article_dublinCore.pdf:PDF},\r\n" +
+            "  groups       = {NO},\r\n" +
+            "  howpublished = {Online},\r\n" +
+            "  keywords     = {Keyword1, Keyword2},\r\n" +
+            "  owner        = {Me},\r\n" +
+            "  review       = {Here are the reviews},\r\n" +
+            "  timestamp    = {2018-02-15},\r\n" +
+            "  url          = {https://www.olly2018.edu},\r\n" +
+            "}";
+
+    private static final String toral2006 = "@InProceedings{Toral2006,\r\n" +
+            "  author     = {Toral, Antonio and Munoz, Rafael},\r\n" +
+            "  title      = {A proposal to automatically build and maintain gazetteers for Named Entity Recognition by using Wikipedia},\r\n" +
+            "  booktitle  = {Proceedings of EACL},\r\n" +
+            "  pages      = {56--61},\r\n" +
+            "  date       = {2006},\r\n" +
+            "  eprinttype = {asdf},\r\n" +
+            "  eventdate  = {2017-05-31},\r\n" +
+            "  owner      = {Christoph Schwentker},\r\n" +
+            "  timestamp  = {2016.11.07},\r\n" +
+            "  url        = {asdfasdfas},\r\n" +
+            "  urldate    = {2017-05-31},\r\n" +
+            "}";
+
+    private static final String vapnik2000 = "@Book{Vapnik2000,\r\n" +
```

Comment:
Please create the BibEntries by hand (using `new BibEntry(), setField`) and not based on the string representation. The XMP test should be as autonomous as possible, especially they shouldn't fail if the BibParser is changed.

Reply:
DONE